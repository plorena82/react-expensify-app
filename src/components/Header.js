
import React from 'react';
import { NavLink} from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <h3><NavLink to="/" activeClassName="is-active"  exact={true}> Go home</NavLink></h3>
        <h3><NavLink to="/create" activeClassName="is-active"> Create Expense</NavLink></h3>
        <h3><NavLink to="/help" activeClassName="is-active"> Help</NavLink></h3>
    </header>
);

export default Header;