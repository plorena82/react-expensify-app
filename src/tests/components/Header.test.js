import React from 'react';
import {shallow} from 'enzyme';
//import toJSON from 'enzyme-to-json'; put in jest.config.json
//import ReactShallowRenderer from 'react-test-renderer/shallow';
import Header from '../../components/Header';


test('should load the header component correctly', ()=>{
    
    const wrapper = shallow(<Header />);//shallow rendering of a component to test
    expect(wrapper).toMatchSnapshot();
    //expect(toJSON(wrapper)).toMatchSnapshot(); toJSON no longer needed as it was added into jest.config.json as snapshotSerializer
    //expect(wrapper.find('h1').text()).toBe('Expensify');

    /*
    const renderer = new ReactShallowRenderer();
    renderer.render(<Header />);//loads the JSX component 
    expect(renderer.getRenderOutput()).toMatchSnapshot();
    */
} );