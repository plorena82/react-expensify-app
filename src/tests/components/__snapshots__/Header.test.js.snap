// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should load the header component correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <h3>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      exact={true}
      to="/"
    >
       Go home
    </NavLink>
  </h3>
  <h3>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      to="/create"
    >
       Create Expense
    </NavLink>
  </h3>
  <h3>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      to="/help"
    >
       Help
    </NavLink>
  </h3>
</header>
`;
